cmd/devicesimulator/main.go

```diff
@@ -87,31 +89,40 @@ func main() {
 	}()
 
 	log.Infoln("running agents")
-	wg := new(sync.WaitGroup)
-	wg.Add(*numDevices)
```

Comment:
note: wg is usually needed in more complex situations, if we properly ensure each goroutine collapses when the context is cancelled waiting on <-ctx.Done() is all we need.